src/Workspaces/Core/MSBuild/MSBuild/Build/ProjectBuildManager.cs

```diff
@@ -28,7 +28,11 @@ internal class ProjectBuildManager
             { PropertyNames.DesignTimeBuild, bool.TrueString },
 
             // this will force CoreCompile task to execute even if all inputs and outputs are up to date
+#if NETCOREAPP
+            { PropertyNames.NonExistentFile, "__NonExistentSubDir__\\__NonExistentFile__" },
```

Comment:
I believe the project system does this even for regular Visual Studio builds, so it seems we would want this in the .NET framework case too.

Reply:
When running on .NET Framework, the property is set here:

https://github.com/microsoft/msbuild/blob/ab9b2f36a5ff7a85f842b205d5529e77fdc9d7ab/src/Tasks/Microsoft.Common.CurrentVersion.targets#L3462-L3470

Reply:
Why isn't that in the .NET Core side of things? Can you add a link to the MSBuilds in the code just so this is captured?

Reply:
> Why isn't that in the .NET Core side of things?

This target works fine in .NET Core. However, triggering this target means setting the `BuildingInsideVisualStudio` property, which has an unwanted side effect of failing elsewhere due to microsoft/msbuild#5045. Since there is no way to trigger this target _without_ setting `BuildingInsideVisualStudio`, we directly set the property instead.

Reply:
Added a comment+link